test/functional/p2p_private_broadcast.py

```diff
@@ -0,0 +1,309 @@
+    "[fc32:17ea:e415:c3bf:9808:149d:b5a2:c9aa]",
+    "[fcc7:be49:ccd1:dc91:3125:f0da:457d:8ce]",
+    "[fcdc:73ae:b1a9:1bf8:d4c2:811:a4c7:c34e]",
+]
+
+class P2PPrivateBroadcast(BitcoinTestFramework):
+    def set_test_params(self):
+        self.disable_autoconnect = False
+        self.num_nodes = 2
+
+    def setup_nodes(self):
+        # Start a SOCKS5 proxy server.
+        socks5_server_config = Socks5Configuration()
+        # self.nodes[0] listens on p2p_port(0),
+        # self.nodes[1] listens on p2p_port(1),
+        # thus we tell the SOCKS5 server to listen on p2p_port(self.num_nodes (which is 2))
+        socks5_server_config.addr = ("127.0.0.1", p2p_port(self.num_nodes))
+        socks5_server_config.unauth = True
+        socks5_server_config.auth = True
+
+        self.socks5_server = Socks5Server(socks5_server_config)
+        self.socks5_server.start()
+
+        ports_base = p2p_port(MAX_NODES) + 15000
+
+        # Add 2 because one feeler and one extra block relay connections may
+        # sneak in between the private broadcast connections.
+        for i in range(NUM_INITIAL_CONNECTIONS + NUM_PRIVATE_BROADCAST_PER_TX + 2):
+            if i == NUM_INITIAL_CONNECTIONS:
+                # Instruct the SOCKS5 server to redirect the first private
+                # broadcast connection from nodes[0] to nodes[1]
+                self.socks5_server.conf.destinations.append({
+                    "to_addr": "127.0.0.1", # nodes[1] listen address
+                    "to_port": p2p_port(1), # nodes[1] listen port
+                    "node": None,
+                    "requested_to_addr": None,
+                })
+            else:
+                # Create a Python P2P listening node and add it to self.socks5_server.conf.destinations[]
+                listener = P2PInterface()
+                listener.peer_connect_helper(dstaddr="0.0.0.0", dstport=0, net=self.chain, timeout_factor=self.options.timeout_factor)
+                listener.peer_connect_send_version(services=P2P_SERVICES)
+                self.network_thread.listen(
+                    p2p=listener,
+                    # Instruct the SOCKS5 server to redirect a connection to this Python P2P listener.
+                    callback=lambda addr, port: self.socks5_server.conf.destinations.append({
+                        "to_addr": addr,
+                        "to_port": port,
+                        "node": None,
+                        "requested_to_addr": None,
+                    }),
+                    addr="127.0.0.1",
+                    port=ports_base + i)
+                # Wait until the callback has been called and it has done append().
+                self.wait_until(lambda: len(self.socks5_server.conf.destinations) == i + 1)
+
+                self.socks5_server.conf.destinations[i]["node"] = listener
+
+        self.extra_args = [
+            [
+                "-peerbloomfilters", # needed to test replies to MEMPOOL
```

Comment:
I don't think is needed, `MEMPOOL` is never tested.

Reply:
Removed.